File: WebDriverTests_Java/WebDriverTests/src/test/java/core/systemPages/SimpleFunctionalTestBase.java
package core.systemPages;


import core.pages.LoginPage;
import core.webDriver.WebDriverCache;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Listeners;

@Listeners({ElementScreenshot.class})
public class SimpleFunctionalTestBase extends WebDriverFunctionalTestBase
{
    @BeforeClass
    @Override
    public void SetUp() {
        super.SetUp();
        LoadPage(LoginPage.class, "user2");
    }

    @AfterClass
    @Override
    public void TearDown()  {
            super.TearDown();
    }

    protected final void ResetDownloadFilesState()
    {
        WebDriverCache.getWebDriver().CleanDownloadDirectory();
    }

    public final void WaitDocuments()  {
        WaitDocuments(90000);
    }

    public final void WaitDocuments(int value) {
        try {
            Thread.sleep(value);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    private core.pages.DefaultPage DefaultPage;
    protected final core.pages.DefaultPage getDefaultPage()
    {
        return DefaultPage;
    }
    private void setDefaultPage(core.pages.DefaultPage value)
    {
        DefaultPage = value;
    }
}

File: WebDriverTests_Csharp/Autotest.Core/Autotests.WebPages/Pages/SimpleFunctionalTestBase.cs
using Autotests.Utilities.WebTestCore;
using Autotests.Utilities.WebTestCore.TestSystem;
using Autotests.WebPages.Pages.PageAdmin;
using Autotests.WebPages.Pages.PageUser;
using System.Threading;
using NUnit.Framework;
using OpenQA.Selenium;

namespace Autotests.WebPages.Pages
{
    [TestFixture]
    public class SimpleFunctionalTestBase : WebDriverFunctionalTestBase
    {
        public override void SetUp()
        {
            base.SetUp();
            LoadPage<LoginPage>("auth/login");
        }

        public override void TearDown()
        {
//            base.TearDown();
            try
            {
                base.TearDown();
            }
            catch (UnhandledAlertException)
            {
                var alert = WebDriverCache.WebDriver.Alert();
                alert.Dismiss();
                base.TearDown();
            }
//            catch (NoSuchElementException)
//            {
//            }
//            finally
//            {
//                
//            }
        }

        public UserHomePage LoginAsUser(string login, string password)
        {
            var enterPage = LoadPage<LoginPage>("auth/login");
            return enterPage.LoginAsUser(login, password);
        }

        public AdminHomePage LoginAsAdmin(string login, string password)
        {
            var enterPage = LoadPage<LoginPage>("auth/login");
            return enterPage.LoginAsAdmin(login, password);
        }

        protected void ResetDownloadFilesState()
        {
            WebDriverCache.WebDriver.CleanDownloadDirectory();
        }

        public LoginPage LoginDefaultPage()
        {
            var partyCreationPage = LoadPage<LoginPage>("auth/login");
            return partyCreationPage;
        }

        public void WaitDocuments(int value = 90000)
        {
            Thread.Sleep(value);
        }

        protected LoginPage LoginPage { get; private set; }
    }
}